src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -160,28 +160,51 @@ partitions_request_iterator validate_partitions(
           return unkown_broker_id_it == partition.replicas->end();
       });
 
+    // Check for undefined topic here instead of outside
+    // validate_partitions because there are only two places to store errors,
+    // the top level and reassignable_partition. The top level should be for
+    // errors that apply to all topics, not a single topic.
     valid_partitions_end = validate_replicas(
       begin,
       valid_partitions_end,
       std::back_inserter(invalid_partitions),
-      error_code::invalid_replication_factor,
-      "Number of replicas does not match the topic replication factor",
+      error_code::unknown_topic_or_partition,
+      "Topic or partition is undefined",
       [&tp_metadata](const reassignable_partition& partition) {
           if (!partition.replicas.has_value()) {
```

Comment:
Why is this condition in the `unknown_topic_or_partition` clause? I figure maybe this should just check if `tp_metadata` is null or not.

Reply:
Ah, that was a mistake. I'll remove.

Reply:
Removing this conditional is leading to compiler issues (the partition param becomes unusued),  and restructuring this check has cascading affects on unit tests. I'll put this edit in an issue and address it in a later PR. But this should not block release.

Reply:
Wait, maybe I'm misunderstanding your point/Q.

An empty replicas vector is a valid request.

Reply:
Sync'd offline. Pushing edits soon. They should not affect test results.